### ISSUE TITLE
.NET - Msi Signature Valid after patching Property->ProductName

### ISSUE THREAD

**Issue by tosmolka** [JetBrains/format-ripper#37]
Hello, I stumbled upon this behavior, I don't think it's expected. Can you please take a look? Thank you. 

Repro steps:
- Copy [data\msi\2dac4b.msi](https://github.com/JetBrains/format-ripper/blob/master/data/msi/2dac4b.msi) to `data\msi\2dac4b_broken_productname.msi`
- Open `data\msi\2dac4b_broken_productname.msi` in Orca
- Change Property->ProductName from `Microsoft Visual Studio 2013 VsGraphics Helper Dependencies` to `XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX`
- Save `data\msi\2dac4b_broken_productname.msi`
- Confirm in Windows Explorer that Digital Certificate is now invalid 
- Add following test case to [net/JetBrains.SignatureVerifier/tests/MsiSignatureVerifierTests.cs](https://github.com/JetBrains/format-ripper/blob/master/net/JetBrains.SignatureVerifier/tests/MsiSignatureVerifierTests.cs) 
```csharp
[TestCase(VerifySignatureStatus.InvalidSignature, "2dac4b_broken_productname.msi")]
```
- Run test and confirm it fails with following error:
```
Expected: InvalidSignature
But was:  Valid
```

Expected behavior:
- Test case succeeds ( == verification returns `VerifySignatureStatus.InvalidSignature` when checking `2dac4b_broken_productname.msi`)

Screenshots:
![image](https://github.com/JetBrains/format-ripper/assets/37370256/6c0f5b1f-9889-4186-9d38-ad08407b0ef1)
<img width="326" alt="image" src="https://github.com/JetBrains/format-ripper/assets/37370256/f9918ea6-ccea-4067-ae0e-add781cffc97">
![image](https://github.com/JetBrains/format-ripper/assets/37370256/ce0162f8-fe82-4923-9dd0-1b354369cbac)

**Comment by Konstantin-Kretov** [comment]
Hi @tosmolka 
Sorry for not informing you in time.
I have added a checksum verification for MSI, PE and Mach-O files in pull requests #38 and #39 which are already merged into master.
Unfortunately, the checksum verification required an API change because the original file stream is needed once again.
I have created AuthenticodeSignatureVerifier and MachOSignatureVerifier classes that include signature and hash verification.
At the moment I have made a 2.1.0-preview.1 release, but I am still working on it.
Please see the updated tests for a usage example.

**Comment by Konstantin-Kretov** [comment]
Hi @tosmolka
We have finally released version 2.1.0 that has proper hashsum verification.

**Comment by tosmolka** [comment]
Thank you @Konstantin-Kretov , much appreciated. I did quickly test the change, it looks comprehensive and it fixes the issue on my side. Thanks you again.
